File: Scala/learn_scala/src/main/scala/com/viper/scala01/Student.scala
package com.viper.scala01

/**
 * @author c1rew    
 * @create 2020-10-03 11:38
 */

// 伴生类
class Student {
    private val name = "Mike"
}

// 伴生对象（静态）
// 伴生对象可以访问伴生类的私有属性
// 创建伴生类对象，需要提供特殊的方法，实现相应的功能
object Student {
    // scala自动识别apply方法，用于创建伴生类对象
    def apply(s:String):Student = new Student()

    def test: Unit ={
        new Student().name
    }
}

File: Java/JDBC/src/main/java/com/viper/utils/Student.java
package com.viper.utils;


import java.util.Date;

/**
 * @author c1rew
 * @date 2020-12-27 15:10
 */

public class Student {
    private String id;
    private String name;
    private Date birth;
    private String sex;

    public Student() {
    }

    public Student(String id, String name, Date age, String sex) {
        this.id = id;
        this.name = name;
        this.birth = age;
        this.sex = sex;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Date getAge() {
        return birth;
    }

    public void setAge(Date birth) {
        this.birth = birth;
    }

    public String getSex() {
        return sex;
    }

    public void setSex(String sex) {
        this.sex = sex;
    }

    @Override
    public String toString() {
        return "Student{" +
                "id='" + id + '\'' +
                ", name='" + name + '\'' +
                ", birth=" + birth +
                ", sex='" + sex + '\'' +
                '}';
    }
}
